RyuSocks/Auth/Packets/UsernameAndPasswordRequest.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Text;
+
+namespace RyuSocks.Auth.Packets
+{
+    public class UsernameAndPasswordRequest : Packet
+    {
+        public byte Version
+        {
+            get
+            {
+                return Bytes[0];
+            }
+            set
+            {
+                Bytes[0] = value;
+            }
+        }
+
+        public byte UsernameLength
+        {
+            get
+            {
+                return Bytes[1];
+            }
+            set
+            {
+                Bytes[1] = value;
+            }
+        }
+
+        public string Username
+        {
+            get
+            {
+                return Encoding.ASCII.GetString(Bytes.AsSpan(2, UsernameLength));
+            }
+            set
+            {
+                ArgumentOutOfRangeException.ThrowIfGreaterThan(value.Length, 0xFF);
+                UsernameLength = (byte)value.Length;
+                Encoding.ASCII.GetBytes(value).CopyTo(Bytes.AsSpan(2, UsernameLength));
+            }
+        }
+
+        public byte PasswordLength
+        {
+            get
+            {
+                return Bytes[2 + UsernameLength];
+            }
+            set
+            {
+                Bytes[2 + UsernameLength] = value;
+            }
+        }
+
+        public string Password
+        {
+            get
+            {
+                return Encoding.ASCII.GetString(Bytes.AsSpan(3 + UsernameLength, PasswordLength));
+            }
+            set
+            {
+                ArgumentOutOfRangeException.ThrowIfGreaterThan(value.Length, 0xFF);
+                PasswordLength = (byte)value.Length;
+                Encoding.ASCII.GetBytes(value).CopyTo(Bytes.AsSpan(3 + UsernameLength, PasswordLength));
+            }
+        }
+
+        public UsernameAndPasswordRequest(byte[] packetBytes)
+        {
+            if (packetBytes.Length is < 4 or > 513)
+            {
+                throw new InvalidOperationException($"IncomingPacket is of wrong length: Expected: > 4 || < 513; Actual {packetBytes.Length}");
+            }
+
+            Bytes = packetBytes;
+            Version = Bytes[0];
+            UsernameLength = Bytes[1];
+
+            try
+            {
+                Username = Encoding.ASCII.GetString(Bytes[2..(2 + UsernameLength)]);
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException($"UsernameLength and actual Username length do not match: {Bytes[1]:X} (throws {e})");
+            }
+
+            PasswordLength = Bytes[2 + UsernameLength];
+
+            try
+            {
+                Password = Encoding.ASCII.GetString(Bytes[(2 + UsernameLength + 1)..((2 + UsernameLength + 1) + PasswordLength)]);
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException($"PasswordLength and actual Password length do not match: {Bytes[1]:X} (throws {e})");
+            }
```

Comment:
```suggestion
```

This is not needed, since `Bytes` is already used by every property to get/set the value.